src/app/collections/[handle]/page.tsx

```diff
@@ -0,0 +1,27 @@
+import CollectionTemplate from "@modules/collections/templates"
+import { Metadata } from "next"
+
+type Props = {
+  params: { handle: string }
+}
+
+const BASEURL = process.env.NEXT_PUBLIC_BASE_URL ?? "https://localhost:3000"
+
+export async function generateMetadata({ params }: Props): Promise<Metadata> {
+  const { collection } = await fetch(
+    `${BASEURL}/collections?handle=${params.handle}`
+  ).then((res) => res.json())
+
+  return {
+    title: `${collection.title} | Acme Store`,
+    description: `${collection.title} collection`,
+  }
+}
+
+export default async function CollectionPage({ params }: Props) {
+  const { collection } = await fetch(
```

Comment:
I'm getting an exception when I open collection details page

Reply:
@fPolic @srindom i can't reproduce this, are you still getting the error after I changed the fallback to http? 
Can you try to add these .env vars?

NEXT_PUBLIC_MEDUSA_BACKEND_URL=http://localhost:9000
NEXT_PUBLIC_BASE_URL=http://localhost:3000

Reply:
Fixed thanks to Frane, the starter runs on port 8000 by default, not 3000